### PR TITLE
Added Google sign in functionality

### PR DESCRIPTION
**Pull request by sicsixdkn**
Added ability to sign in using Google.
- Restyled login page to include Google sign in button
- Added login functionality to LoginActivity

I experimented with separating out the login functionality to a Service class, however scrapped this and instead generate and use the intents locally. I'm open to suggestions here, given I'm very very new to Java/Kotlin/Android.

Also, this won't work right now. It returns a status code 10 which seems to be a configuration issue with Firebase/Google requiring signed keys from the application before allowing signin - see https://stackoverflow.com/questions/49450140/google-signin-api-exception-10

Without access to the consoles for either Firebase for the app I can't fully test this.

### CODE REVIEW

File: app/src/main/res/layout/activity_login.xml [hunk 1]
```diff
@@ -48,13 +50,30 @@
...
         android:fontFamily="@font/poppins_bold"
         android:text="@string/login_register_btn"
         android:textAllCaps="false"
         android:textColor="@color/white"
         app:cornerRadius="24dp"
+        app:layout_constraintBottom_toTopOf="@+id/loginGoogleBtn"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
+
+    <Button
```
Comment (el-veee) [inline app/src/main/res/layout/activity_login.xml]: You can replace `Button` with `com.google.android.gms.common.SignInButton` so the component looks more standard for Google sign in
Screenshot 2023-11-24 at 12 54 40 pm

File: app/src/main/java/deakin/gopher/guardian/view/general/LoginActivity.kt [hunk 2]
```diff
@@ -19,16 +28,25 @@ import deakin.gopher.guardian.services.EmailPasswordAuthService
...
+    private lateinit var gsoClient: GoogleSignInClient
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_login)
         val mEmail: EditText = findViewById(R.id.Email)
         val mPassword: EditText = findViewById(R.id.password)
         val progressBar: ProgressBar = findViewById(R.id.progressBar)
         val loginButton: Button = findViewById(R.id.loginBtn)
+        val loginGoogleButton: Button = findViewById(R.id.loginGoogleBtn)
```
Comment (el-veee) [inline app/src/main/java/deakin/gopher/guardian/view/general/LoginActivity.kt]: Cast to a `SignInButton` from `com.google.android.gms.common.SignInButton` for the standard button type

**Review by el-veee — COMMENTED** [review]
Would be good to see the Google-style button but otherwise looks and works as expected. I am happy for this task to be completed and another task to be created for the backend implementation when we get Firebase access

File: app/src/main/res/layout/activity_login.xml (hunk 1 above)
Comment (sicsixdkn) [reply app/src/main/res/layout/activity_login.xml]: Seems like a good idea, changed!

File: app/src/main/java/deakin/gopher/guardian/view/general/LoginActivity.kt (hunk 2 above)
Comment (sicsixdkn) [reply app/src/main/java/deakin/gopher/guardian/view/general/LoginActivity.kt]: Yeah crashes without this apparently. I would have thought Button would be the superclass of SignInButton and the upcasting would be handled automatically... apparently not!

**Review by el-veee — APPROVED** [review]

**Pull request merged**